### PR TITLE
Revert iphelper workaround

### PR DESCRIPTION
**Pull request by jterry75**
### Summary

Reverts: #3328 

Adds an extra logging line to better signify when the LinkName fails to resolve because the ENI is not match by MAC address.

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by rawahars — APPROVED** [review]

File: agent/api/eni/eni.go [hunk 1]
```diff
@@ -214,6 +214,9 @@ func (eni *ENI) GetLinkName() string {
 				break
 			}
 		}
+		if eni.LinkName == "" {
```
Comment (singholt) [inline agent/api/eni/eni.go:217]: is this the same check as on L203 and the same error log line as L207?

Comment (jterry75) [reply agent/api/eni/eni.go:217]: Not exactly. What we found is that we can in the above for loop, loop through all interfaces, not match any, and then return "" with no logging. So we verify that indeed one was matched or else we at least log that it wasnt so we understand later why the CNI fails to find an empty adapter LinkName

Comment (singholt) [reply agent/api/eni/eni.go:217]: nvm, I think this is to check if link name is empty even after the loop on L210.

nit: could you add a comment on this?

**Review by singholt — DISMISSED** [review]

**Review by vsiddharth — APPROVED** [review]

File: agent/api/eni/eni.go (hunk 1 above)
Comment (jterry75) [reply agent/api/eni/eni.go:220]: Sure thing. Done.

**Review by singholt — APPROVED** [review]

**Review by Realmonia — APPROVED** [review]

**Pull request merged**